Patch-Source: https://github.com/jthornber/thin-provisioning-tools/pull/249
--
From 859bb543f76df5088a22e2b7f1d6e2cb22bcd4f2 Mon Sep 17 00:00:00 2001
From: psykose <alice@ayaya.dev>
Date: Wed, 15 Feb 2023 18:23:23 +0000
Subject: [PATCH] [cache_metadata_size] fix build on 32-bit

(usize * usize) as u64 does not cast early enough; the multiplication
can still overflow
---
 src/cache/metadata_size.rs | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cache/metadata_size.rs b/src/cache/metadata_size.rs
index 2bceb15..ec3998a 100644
--- a/src/cache/metadata_size.rs
+++ b/src/cache/metadata_size.rs
@@ -35,7 +35,7 @@ pub fn metadata_size(opts: &CacheMetadataSizeOptions) -> Result<u64> {
     let hint_size = opts.nr_blocks * (opts.max_hint_width as u64 + HINT_OVERHEAD_PER_BLOCK);
 
     let mut size = TRANSACTION_OVERHEAD + mapping_size + hint_size;
-    size = std::cmp::min(size, (MAX_METADATA_BLOCKS * BLOCK_SIZE) as u64);
+    size = std::cmp::min(size, MAX_METADATA_BLOCKS as u64 * BLOCK_SIZE as u64);
 
     Ok(size)
 }
-- 
2.39.2

